### PR TITLE
Address tech debt

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #136 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Refactor the AuroIcon component to utilize a new runtime utility for component registration and tag renaming, simplifying the codebase and improving maintainability. Update documentation and tests to align with these changes, and introduce a new ESLint configuration for consistent code quality. Remove outdated scripts to reduce technical debt.

Enhancements:
- Refactor the component registration process to use a new runtime utility for handling custom element names.
- Remove redundant code by replacing the handleCustomTagName function with a method from the runtime utility.
- Introduce a new runtime utility instance in the AuroIcon component for handling component tag renaming.

Build:
- Add a new ESLint configuration file to manage linting rules and configurations.

Documentation:
- Update documentation to reflect changes in the component registration process using the new runtime utility.

Tests:
- Modify tests to accommodate the new component registration process using the runtime utility.

Chores:
- Remove obsolete scripts related to documentation generation and static styles.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,15 +1,4 @@
...
- */
- const registerComponent = (name = 'custom-icon') => {
-  // alias definition
-  if (!customElements.get(name)) {
-    customElements.define(name, class extends AuroIcon {});
-  }
-}
-
-export { registerComponent };
+RuntimeUtils.default.prototype.registerComponent('custom-icon', AuroIcon);
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply index.js:4]: Not addressing this as of right now.

File: test/auro-icon.test.js [hunk 2]
```diff
@@ -8,10 +8,10 @@
 /* eslint-disable no-undef */
 import { fixture, html, expect, waitUntil } from '@open-wc/testing';
 import sinon from 'sinon';
-// import '../src/auro-icon.js';
+import { AuroIcon } from '../src/auro-icon.js';
+import * as RuntimeUtils from '@aurodesignsystem/auro-library/scripts/utils/runtimeUtils.mjs';
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply test/auro-icon.test.js:12]: This test is not useful for us.

File: test/auro-icon.test.js [hunk 3]
```diff
@@ -277,7 +277,7 @@ describe('auro-icon', () => {
   it('auro-icon can be registered as a custom element name', async () => {
     const customElementName = 'custom-icon';
 
-    registerComponent(customElementName);
+    RuntimeUtils.default.prototype.registerComponent('custom-icon', AuroIcon);
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply test/auro-icon.test.js:280]: Great idea that will be addressed later.

File: docs/partials/index.md [hunk 4]
```diff
@@ -69,8 +69,10 @@ There are two important parts of every Auro component. The <a href="https://deve
 To protect from versioning conflicts with other instances of the component being loaded, it is recommended to use our `registerComponent(name)` method and pass in a unique name.
 
 ```js
-import './node_modules/@aurodesignsystem/auro-icon';
-registerComponent('custom-icon');
+import { AuroIcon } from './src/auro-icon.js';
+import * as RuntimeUtils from '@aurodesignsystem/auro-library/scripts/utils/runtimeUtils.mjs';
+
+RuntimeUtils.default.prototype.registerComponent('custom-icon', AuroIcon);
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply docs/partials/index.md:75]: Not adding comments at this time.

File: demo/index.md [hunk 5]
```diff
@@ -85,8 +85,10 @@ There are two important parts of every Auro component. The <a href="https://deve
 To protect from versioning conflicts with other instances of the component being loaded, it is recommended to use our `registerComponent(name)` method and pass in a unique name.
 
 ```js
-import './node_modules/@aurodesignsystem/auro-icon';
-registerComponent('custom-icon');
+import { AuroIcon } from './src/auro-icon.js';
+import * as RuntimeUtils from '@aurodesignsystem/auro-library/scripts/utils/runtimeUtils.mjs';
+
+RuntimeUtils.default.prototype.registerComponent('custom-icon', AuroIcon);
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply demo/index.md:91]: Not adding comments at this time.

File: test/auro-icon.test.js [hunk 6]
```diff
@@ -8,10 +8,10 @@
 /* eslint-disable no-undef */
 import { fixture, html, expect, waitUntil } from '@open-wc/testing';
 import sinon from 'sinon';
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply test/auro-icon.test.js:10]: Will add tests in the future.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**